### ISSUE TITLE
Implement parent selector & and selector functions

### ISSUE THREAD

**Issue by robocoder** [leafo/scssphp#310]
- [x] parent selector `&`, e.g.,
```
.foo.bar .baz.bang, .bip.qux {
  $selector: &;
}
```

where the value of $selector is now ((".foo.bar" ".baz.bang"), ".bip.qux").
- [ ] selector-nest()
- [ ] selector-append()
- [ ] selector-extend()
- [ ] selector-replace()
- [ ] selector-unify()
- [ ] is-superselector()
- [ ] simple-selectors()
- [ ] selector-parse()

**Comment by Cerdic** [comment]
For the record, what we have right now for the input
```
.foo.bar .baz.bang, .bip.qux {
  $selector: &;
  content: "#{$selector}";
}
```
is
```
.foo.bar .baz.bang, .bip.qux {
  content: ".foo.bar.baz.bang, .bip.qux"; }
```

**Comment by Cerdic** [comment]
parent selector is implemented in #693

**Comment by robocoder** [comment]
I've closed this for now as there is now basic support. However, there are some parent selector test cases in tests/scss_test.rb that are failing which will be addressed as devs (periodically) work towards increasing compatibility with the reference implementation.
